tutorials-v5/time-evolution/024_v5_paper-floquet-speed-test.md

```diff
@@ -0,0 +1,332 @@
+        state_list = [basis(2, 1)] * (N - 1)
+        state_list.append(basis(2, 0))
+        psi0 = tensor(state_list)
+
+        # Defining expectation operator
+        e_ops = sz_list
+        return H, psi0, e_ops, args
+```
+
+```python
+H, psi0, e_ops, args = setup_Ising_drive(N, g0, J0, A, omega)
+```
+
+```python
+# Simulation parameters
+N_T = 10
+dt = 0.01
+tlist = np.arange(0, N_T * T, dt)
+tlist_0 = np.arange(0, T, dt)  # One period tlist
+
+options = {"progress_bar": False, "store_floquet_states": True}
+```
+
+```python
+computational_time_floquet = np.ones(tlist.shape) * np.nan
+computational_time_sesolve = np.ones(tlist.shape) * np.nan
+
+expect_floquet = np.zeros((N, len(tlist)))
+expect_sesolve = np.zeros((N, len(tlist)))
+```
+
+```python
+# Running the simulation
+for n, t in enumerate(tlist):
+    # Floquet basis
+    # --------------------------------
+    tic_f = time.perf_counter()
+    if t < T:
+        # find the floquet modes for the time-dependent hamiltonian
+        floquetbasis = FloquetBasis(H, T, args)
+    else:
+        floquetbasis = FloquetBasis(H, T, args, precompute=tlist_0)
+
+    # Decomposing inital state into Floquet modes
+    f_coeff = floquetbasis.to_floquet_basis(psi0)
+    # Obtain evolved state in the original basis
+    psi_t = floquetbasis.from_floquet_basis(f_coeff, t)
+    p_ex = expect(e_ops, psi_t)
+    toc_f = time.perf_counter()
+
+    # Saving data
+    computational_time_floquet[n] = toc_f - tic_f
+
+    # sesolve
+    # --------------------------------
+    tic_f = time.perf_counter()
+    output = sesolve(H, psi0, tlist[: n + 1], e_ops=e_ops, args=args)
+    p_ex_r = output.expect
+    toc_f = time.perf_counter()
+
+    # Saving data
+    computational_time_sesolve[n] = toc_f - tic_f
+
+    for i in range(N):
+        expect_floquet[i, n] = p_ex[i]
+        expect_sesolve[i, n] = p_ex_r[i][-1]
+```
+
+```python
+fig, axs = plt.subplots(1, 2, figsize=(12, 5))
+ax = axs[0]
+axs[0].plot(tlist / T, computational_time_floquet, "-")
+axs[0].plot(tlist / T, computational_time_sesolve, "--")
+
+axs[0].set_yscale("log")
+axs[0].set_yticks(np.logspace(-3, -1, 3))
+lg = []
+for i in range(N):
+    axs[1].plot(tlist / T, np.real(expect_floquet[i, :]), "-")
+    lg.append(f"n={i+1}")
+for i in range(N):
+    axs[1].plot(tlist / T, np.real(expect_sesolve[i, :]), "--", color="black")
+lg.append("sesolve")
+
+axs[0].set_xlabel(r"$t \, / \, T$")
+axs[1].set_xlabel(r"$t \, / \, T$")
+axs[0].set_ylabel(r"$Computational \; Time,\; [s]$")
+axs[1].set_ylabel(r"$\langle \sigma_z^{{(n)}} \rangle$")
+axs[0].legend(("Floquet", "sesolve"))
+axs[1].legend(lg)
+xticks = np.rint(np.linspace(0, N_T, N_T + 1, endpoint=True))
+axs[0].set_xticks(xticks)
+axs[0].set_xlim([0, N_T])
+axs[1].set_xticks(xticks)
+axs[1].set_xlim([0, N_T])
+txt = f"$N={N}$, $g_0={g0}$, $J_0={J0}$, $A={A}$"
+axs[0].text(0.0, 1.05, txt, transform=axs[0].transAxes)
+
+plt.show()
+```
+
+## About
+
+```python
+about()
+```
+
+## Testing
+
```

Comment:
I think expect_floquet  and expect_sesolve should be pretty close, so can be used?

Reply:
galilabarias.marc@aist.go.jp